File: trunk/src/ruby/damagecontrol/cruisecontrol/CruiseControlBridgeTest.rb
require 'test/unit'
require 'damagecontrol/cruisecontrol/CruiseControlBridge'

module DamageControl

	class CruiseControlBridgeTest < Test::Unit::TestCase
		def test_call_cc_bridge_with_fake_source_control
			bridge = CruiseControlBridge.new
			bridge.sourcecontrol = "damagecontrol.FakeSourceControl"
			bridge.parameters = { "type" => "ruby_type" }
			now = Time.now
			modification = bridge.modifications(now, now)[0]
			assert_equal("-ruby_type", modification.type) # TODO this is actually wrong!
			assert_equal(now.to_s, modification.modified_time.to_s)
			assert_equal("revision", modification.revision)
			assert_equal("comment", modification.comment)
			assert_equal("emailAddress", modification.email_address)
			assert_equal("fileName", modification.file_name)
			assert_equal("folderName", modification.folder_name)
		end
	end

end

File: trunk/src/test/damagecontrol/CruiseControlBridgeTest.java
package damagecontrol;

import junit.framework.TestCase;
import net.sourceforge.cruisecontrol.Modification;

import java.util.Date;

public class CruiseControlBridgeTest extends TestCase {
    public void test() throws Exception {
        CruiseControlBridge bridge = new CruiseControlBridge();
        Date lastbuild = new Date();
        Date now = new Date();
        bridge.init(new String[] { "damagecontrol.FakeSourceControl",
                                   "-type", "type",
                                   "-lastbuild", CruiseControlBridge.DATE_FORMAT.format(lastbuild),
                                   "-now", CruiseControlBridge.DATE_FORMAT.format(now) });
        assertDateEquals(lastbuild, bridge.lastbuild);
        assertDateEquals(now, bridge.now);
        assertTrue(bridge.sourceControl instanceof FakeSourceControl);
        FakeSourceControl sourceControl = (FakeSourceControl) bridge.sourceControl;
        Modification modification = (Modification) sourceControl.getModifications(bridge.lastbuild, bridge.now).get(0);
        assertEquals("type", modification.type);
        assertEquals(bridge.now, modification.modifiedTime);
    }

    private void assertDateEquals(Date expected, Date actual) {
        assertEquals(CruiseControlBridge.DATE_FORMAT.format(expected), CruiseControlBridge.DATE_FORMAT.format(actual));
    }

}
